airflow/api_fastapi/execution_api/schemas.py

```diff
@@ -119,3 +119,30 @@ class TIHeartbeatInfo(BaseModel):
 
     hostname: str
     pid: int
+
+
+class ConnectionResponse(BaseModel):
+    """Connection serializer for responses."""
+
+    conn_id: str
+    conn_type: str
+    description: str | None
+    host: str | None
+    schema_: str | None = Field(alias="schema")
```

Comment:
90% of this schema is redundant with the one from the public API.

Should we factorize them with the same `Base`, or should we duplicate all ?

First intuition would be for factorization.

Reply:
Some of the key differences for this one are that we aren't redacting passwords or contents of "extra".  and the good thing is we can change response model whenever we want since it is internal so IMO while some of it is redundant, it is still fine